frontends/concrete-python/concrete/fhe/compilation/server.py

```diff
@@ -416,10 +440,41 @@ def run(
                 raise ValueError(message)
 
             if not isinstance(arg, Value):
-                message = f"Expected argument {i} to be an fhe.Value but it's {type(arg).__name__}"
-                raise ValueError(message)
+                if i not in self._clear_input_indices[function_name]:
+                    message = (
+                        f"Expected argument {i} to be an fhe.Value "
+                        f"but it's {friendly_type_format(type(arg))}"
+                    )
+                    raise ValueError(message)
+
+                exporter = SimulatedValueExporter.new(
```

Comment:
Well done, I don't thought about using the simulated value exporter, indeed that works but a comment to explain why we use it and why that works can be a plus.

Reply:
Sure.